tests/core/sinks/test_validation.py

```diff
@@ -39,3 +41,62 @@ def test_validate_record():
     )
     assert updated_record["missing_datetime"] == "2021-01-01T00:00:00+00:00"
     assert updated_record["invalid_datetime"] == "9999-12-31 23:59:59.999999"
+
+
+@pytest.fixture
+def bench_sink() -> BatchSinkMock:
+    target = TargetMock()
+    return BatchSinkMock(
+        target,
+        "users",
+        {
+            "type": "object",
+            "properties": {
+                "id": {"type": "integer"},
+                "created_at": {"type": "string", "format": "date-time"},
+                "updated_at": {"type": "string", "format": "date-time"},
+                "deleted_at": {"type": "string", "format": "date-time"},
+            },
+        },
+        ["id"],
+    )
+
+
+@pytest.fixture
+def bench_record():
+    return {
+        "id": 1,
+        "created_at": "2021-01-01T00:08:00-07:00",
+        "updated_at": "2022-01-02T00:09:00-07:00",
+        "deleted_at": "2023-01-03T00:10:00.0000",
+    }
+
+
+def test_bench_parse_timestamps_in_record(benchmark, bench_sink, bench_record):
+    """Run benchmark for Sink method _parse_timestamps_in_record."""
+    number_of_runs = 10000
+
+    sink: BatchSinkMock = bench_sink
+    record: dict = bench_record
+
+    def run_parse_timestamps_in_record():
+        for _ in range(number_of_runs):
+            _ = sink._parse_timestamps_in_record(
+                record, sink.schema, sink.datetime_error_treatment
+            )
```

Comment:
> @edgarrmondragon I am not quite sure why `test_validate_record` is passing while `test_bench_parse_timestamps_in_record` and `test_bench_validate_and_parse` are failing. I would think all three would pass or fail the same way. When you have some time would you please review the `codspeed` errors.

@BuzzCutNorman I think date-time strings are parsed _in place_, so the first time `_parse_timestamps_in_record` runs they're converted, then the second call fails because they're no longer strings. Does that make sense?

Reply:
@edgarrmondragon that does make sense.  I just tried using `copy.copy`, `copy.deepcopy` and also moving them inside the function.  I am still seeing the error.  I think still not correctly isolating `record` from getting updated between bench tests.

Reply:
I think I see it now.

Reply:
@edgarrmondragon Please take a look and let me know what you think. 😊